### ISSUE TITLE
Add Homepage Rotator

### ISSUE THREAD

**Issue by willthemoor** [Ecotrust/ocs-wp#9]
<img width="692" alt="screen shot 2015-09-06 at 6 05 44 pm" src="https://cloud.githubusercontent.com/assets/54206/9707594/f65c9920-54c1-11e5-9c2a-d275a3e7e4f2.png">

* Add a CMB repeater for the page "Home" (ID: 22) with three fields:
  * Success Story content type selector
  * Headline
  * Caption
  * No limit to the total number of slides
* CSS: Homepage container will need an Oregon outline mask (SVG or PNG) so they don't need to be created by hand per story
* JS/CSS for actual rotator. Site has Bootstrap installed but whatever works.
* *Nice to have/Low Priority*: When a success story is selected, check to see if that success story has a featured image associated with it and that it's of an appropriate size. If not, warn the user within the homepage editor.

**Comment by willthemoor** [comment]
When outputting success stories in any template it needs to have a #success-story hash around the container.

ADD: In the admin UI for this homepage rotator, the user will select which OCS page it will link to (presumably the page with this success story). We'll append #success-story to the url via code.

**Comment by dnseminara** [comment]
setup for mask and carousel is *mostly* implemented locally

once #35 is complete, I can setup the logic to loop through images and their attribution/name/caption fields

![screen shot 2015-12-02 at 1 57 34 pm](https://cloud.githubusercontent.com/assets/6739893/11545563/b4107b20-98fc-11e5-8333-8dc442d944e2.png)


![screen shot 2015-12-02 at 1 54 33 pm](https://cloud.githubusercontent.com/assets/6739893/11545445/3fecf0de-98fc-11e5-899d-766ad3de1dac.png)

@ldford - let me know if you have any further styling suggestions - left and right arrows only appear on image carousel hover

**Comment by dnseminara** [comment]
live dev site shows no styling - probably just needs a friendly ```gulp clean ``` and a ```gulp build``` to load styling and jquery

**Comment by willthemoor** [comment]
Admin fields for the homepage are added. Also added the PHP code to the homepage template to loop through them.

Couple of issues:

1. In Firefox (mac, latest) it only masks on the first slide. It briefly shows the mask when switching slides.
2. In all browsers I tried this in, the mask isn't full (the outer edges are cropped). I see it working in the screenshots above though! I'm not sure what's different. I tried reducing the window size to make sure it wasn't related to the size of the original image (would should probably be accounted for in code anyway) but that didn't work either. Did I mess up the code? 
3. This isn't so much of an issue but we told the client that the images would rotate while the Oregon outline remained fixed. Apologies for not mentioning this in the original ticket. Think they'd be fine with this but I actually think a cross-fade would like best. @ldford? 

<img width="712" alt="screen shot 2015-12-05 at 11 12 54 pm" src="https://cloud.githubusercontent.com/assets/54206/11612215/2d25ca14-9ba6-11e5-8e24-cfb42547dfa3.png">

**Comment by dnseminara** [comment]
Got it! Revamped the structure so the images slide into the mask (which stays fixed), rather than having the image already masked on window entrance.

tested in Moz, Chrome and IE 11.

**Comment by ldford** [comment]
@dnseminara Looks great. Could you please let me know what the minimum image size requirements are for this? Realize that ODFW will likely need to upload in admin: link + caption + image since the image on the page they are linking to will likely not be big enough to display nicely on homepage.

**Comment by willthemoor** [comment]
Nice word Mr. Drew. I've updated dev with the latest.

**Comment by willthemoor** [comment]
Ack, sorry. One more: there is what appears to be some horizontal noise on the bottom of the png.

![screen shot 2015-12-09 at 11 27 55 am](https://cloud.githubusercontent.com/assets/54206/11696183/ecdfb072-9e67-11e5-914c-99ae5ea085c1.png)
![screen shot 2015-12-09 at 11 27 17 am](https://cloud.githubusercontent.com/assets/54206/11696184/ed02047e-9e67-11e5-8015-6dcc0569f024.png)

Think it would also be good if the whole thing was vertically centered but that's minor.

**Comment by dnseminara** [comment]
Aha! Nice catch, for some reason that was actually the bottom boundary Oregon had when I created the png - weird. Modifying that now.

@ldford - the images should ideally be wider than they are tall, but I've implemented it so that most images will work. The mask is roughly 740px by 350px and I think something similar in ratio should be followed to get the best view within the mask.

**Comment by ldford** [comment]
@dnseminara Hate reopening closed issues, but realized I forgot about the vision for the homepage - prior to the rotator actually kicking in. Page 1 of these comps is the official "homepage" that people will land on every time they go to the URL or click on "Conserve" - it's the cover of the 2006 OCS. Then the rotator begins (page 2). There is no copy associated with it and no link. I tried to upload a jpeg of the cover to try it out within the admin you built, but I'm getting an error from the media library. And it looks as though links are required with each image. I don't think it would be a big deal for the ODFW team for the OCS cover to be number 1 in the rotator - as long as it's always the first one people land on and if there is no associated text or link, it still looks good. Thoughts? 
[OCS homepage.pdf](https://github.com/Ecotrust/ocs-wp/files/58860/OCS.homepage.pdf)

**Comment by willthemoor** [comment]
FWIW, I think it should just go in as an image (presuming that's working again) and it can link to the introduction. That way they can swap it out later if they want to (vs us hardcoding it in as a "pre-rotator" image).

@dnseminara I don't think I built the loop very defensively; you'll want to add something to not out put the heading/caption (and not error) if they're blank.

**Comment by ldford** [comment]
Yes - just connected with Drew about this. All good. Keeping ticket open because links not linking.
